pkg/plugin/processor/builtin/convertField.go

```diff
@@ -0,0 +1,156 @@
+// Copyright © 2024 Meroxa, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//go:generate paramgen -output=convertField_paramgen.go convertFieldConfig
+
+package builtin
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+
+	"github.com/conduitio/conduit-commons/opencdc"
+	sdk "github.com/conduitio/conduit-processor-sdk"
+	"github.com/conduitio/conduit/pkg/foundation/cerrors"
+)
+
+type convertField struct {
+	referenceResolver sdk.ReferenceResolver
+	config            convertFieldConfig
+
+	sdk.UnimplementedProcessor
+}
+
+func newConvertField() *convertField {
+	return &convertField{}
+}
+
+type convertFieldConfig struct {
+	// Field is the target field, as it would be addressed in a Go template (e.g. `.Payload.After.foo`).
+	// you can only convert fields that are under .Key and .Payload, and said fields should contain structured data.
+	Field string `json:"field" validate:"required,regex=^.(Payload|Key).*"`
```

Comment:
We need to escape `.`, otherwise we are matching any character at the start (e.g. `aPayload` is currently valid).
```suggestion
	Field string `json:"field" validate:"required,regex=^\.(Payload|Key).*"`
```

Reply:
hmm looks like we can't have that in a tag, golangci-lint is complaining, and I tried a test with `sPayload` and `regex=^\.(Payload|Key).*`, the validation still passed, so it's not even considering the existence of the `escape`.
the validation passed but ofc the `Configure()` method failed, because of the resolver.
should we just keep it as is, and we know it will fail when trying to get the reference? or move the validation back to `Configure()` manually?
this looks like it would be a bigger problem for paramgen itself, cuz regexes usually have `escape` characters

Reply:
Hmm, it's interesting that the linter complains. What does it say? Dumb question, but did you run paramgen again after updating the regex? The regex looks correct, I'm surprised it didn't work.:S

Reply:
Maha's right, because this is a string we need to escape the `\` itself, so this should work:
```suggestion
	Field string `json:"field" validate:"required,regex=^\\.(Payload|Key).*"`
```